documents/service.go

```diff
@@ -58,10 +58,10 @@ type Service interface {
 	ReceiveAnchoredDocument(ctx context.Context, model Model, senderID []byte) error
 
 	// Create validates and persists Model and returns a Updated model
-	Create(ctx context.Context, model Model) (Model, uuid.UUID, error)
+	Create(ctx context.Context, model Model) (Model, uuid.UUID, chan bool, error)
```

Comment:
not sure why we need to have the chan here since this is ignored mostly. I would prefer not having this part of the interface

Reply:
its needed so that any code that uses this service function can wait on the update in an async way or the proper go way, instead of polling db. We allow polling of API for outside users but no need to do that inside the node. Therefore this has to be a convention we follow for all aync background funcs.